Вариант 16.4/Вариант 16.4/Program.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+
+namespace Cycle
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите значение n");
+            int n = int.Parse(Console.ReadLine());
+            int c = 1;
+
+            for (int i = 1; i != n; i++)
+            {
+                c = c * 10;
+            }
+
+            Console.WriteLine("Все простые n-значные числа:");
+
+            for (int i = c; i != c*10; i++)
+            {
+                bool flag = true;
+                for (int j = 2; j != i-1; j++)
+                {
+                    if (i % j == 0)
+                        flag = false;
+                }
```

Comment:
Вот тут можно оптимизировать. Когда уже обнаружено, что число не простое, внутренний цикл можно прервать.
```suggestion
                bool flag = true;
                
                for (int j = 2; j != i-1; j++)
                    if (i % j == 0)
                    {
                        flag = false;
                        break;
                    }
```